### DISCUSSION TITLE
Rule based sampling - opentelemetry-java-contrib

### DISCUSSION THREAD

**Discussion by chetan5375** [open-telemetry/opentelemetry-java#5629] [Q&A]
Hello, 

I have a requirement to filter out spans based on some predefined rules like, span kind, span name and/or some span attributes. I came across https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/samplers/src/main/java/io/opentelemetry/contrib/sampler/RuleBasedRoutingSampler.java  sampler from opentelemetry-java-contrib project which supports SamplingRules based on span kind and span attributes. 

Is there a way to use/configure this Sampler based to respect span name based sampling rules as well?

Or is there any out of the box head based Sampler available which we can use to filter out spans based on span kind / name/attribute values?

Please suggest.

Thanks,
Chetan

**Comment by mateuszrzeszutek** [comment]
Hey @chetan5375 ,

> Is there a way to use/configure this Sampler based to respect span name based sampling rules as well?
> 
> Or is there any out of the box head based Sampler available which we can use to filter out spans based on span kind / name/attribute values?

There is not; the rule based sampler only supports filtering by attributes.
However, you can simply implement the [`Sampler`](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java) interface yourself, and configure the SDK to use your own implementation.
